lib/kafka-consumer.js

```diff
@@ -413,6 +413,52 @@ KafkaConsumer.prototype.unsubscribe = function() {
   return this;
 };
 
+/**
+ * Read a number of messages from a specific topic and partition.
+ *
+ * Can be useful if the consume performance differs per partition. Consuming
+ * per partition could prevent slow performance on one partition from affecting
+ * the consumption of other partitions.
+ *
+ * To select the right partition it is required to set a topic param, because a
+ * consumer can be subscribed to multiple topics.
+ *
+ * @param {string} topic - Name of topic to read
+ * @param {number} partition - Identifier of partition to read
+ * @param {KafkaConsumer~readCallback} cb - Callback to return when work is done.
+ * @param {object} options - Optional settings for consume
+ */
+ KafkaConsumer.prototype.consumeFromToppar = function(topic, partition, cb, options = {}) {
+  if(typeof topic !== 'string') {
+    throw new TypeError('Topic must be a string');
+  }
+  if(typeof partition !== 'number') {
+    throw new TypeError('Partition must be a number');
+  }
+  if(typeof cb !== 'function') {
+    throw new TypeError('Callback must be a function');
+  }
+  if(typeof options !== 'object') {
+    throw new TypeError('Options must be an object');
+  }
+  if(typeof options.numberOfMessages !== 'number' && typeof options.numberOfMessages !== 'undefined') {
+    throw new TypeError('options.numberOfMessages must be a number or undefined');
+  }
+  if(typeof options.timeoutMs !== 'number' && typeof options.timeoutMs !== 'undefined') {
+    throw new TypeError('options.timeoutMs must be a number or undefined');
+  }
+  if(typeof options.onlyApplyTimeoutToFirstMessage !== 'boolean' && typeof options.onlyApplyTimeoutToFirstMessage !== 'undefined') {
+    throw new TypeError('options.onlyApplyTimeoutToFirstMessage must be a boolean or undefined');
+  }
```

Comment:
That could introduce some unwanted behavior when `options.onlyApplyTimeoutToFirstMessage === false` or `options.timeoutMs === 0`, which are perfectly fine values.

Reply:
Yes, for the boolean options that's true. For the numeric not necessarily because you don't expect non-numeric values there, other than `undefined` ofc..

Reply:
What do you propose? If we change `typeof options.timeoutMs !== 'number' && options.timeoutMs !== undefined` to `typeof options.timeoutMs !== 'number' && !options.timeoutMs`. Then an input of `0` will result in an error. While `"0"` will not do that. That seems undesirable?

Reply:
As just discussed, fine to keep it like this as it's a more sound check in any case. When/if you want to PR to upstream, they can see whether they want to keep this or change in-line with the style around it.